pkg/git/Bean.go

```diff
@@ -56,13 +61,39 @@ type GitCommit struct {
 	Message   string
 	Changes   []string          `json:",omitempty"`
 	FileStats *object.FileStats `json:",omitempty"`
+	PrData	  *PrData			`json:"prData"`
 }
+
+type PrData struct {
+	Id					int 	`json:"id"`
+	PrTitle        		string  `json:"prTitle"`
+	PrUrl        		string	`json:"prUrl"`
+	SourceBranchName    string	`json:"sourceBranchName"`
+	SourceBranchHash    string	`json:"sourceBranchHash"`
+	TargetBranchName    string	`json:"targetBranchName"`
+	TargetBranchHash    string	`json:"targetBranchHash"`
+	AuthorName		    string	`json:"authorName"`
+	LastCommitMessage	string	`json:"lastCommitMessage"`
+	PrCreatedOn   		time.Time `json:"prCreatedOn"`
+	PrUpdatedOn   		time.Time `json:"prUpdatedOn"`
+}
+
+
 type CommitMetadataRequest struct {
 	PipelineMaterialId int    `json:"pipelineMaterialId"`
 	GitHash            string `json:"gitHash"`
 	GitTag             string `json:"gitTag"`
 }
 
+type LatestCommitMetadataRequest struct {
```

Comment:
I dont see any field which restricts it to Latest, so we should not have Latest in the name of the struct.. also cant we combine it with the previous struct i.e; CommitMetadataRequest, ll there be some logical inconsistency or purpoose will not be clear if we combine them?

Reply:
ok.
CommitMetadataRequest is having gitHash in the request object.
LatestCommitMetadataRequest is having branchName in the request object.

Implementation is different for these two. The later one will fetch the last commit info for that branch name. 

But ofcourse we can merge this.

Reply:
merged